### PR TITLE
Hukam multi shabad regression

### PR DESCRIPTION
**Pull request by ManjotS**
this was really complicated and im not confident enough in js to know if I did this right. but it works.

### CODE REVIEW

File: api/controllers/shabads.js [hunk 1]
```diff
@@ -438,64 +437,91 @@ const getShabad = (ShabadIDQ, sinceDate = null) =>
...
+      }
+    });
+    return {
+      previous,
+      next,
+    };
+  } catch (err) {
+    error(err);
+  } finally {
+    if (conn) conn.end();
```
Comment (tsingh777) [inline api/controllers/shabads.js]: nit - prefer `if` with curly brackets

if you don't want to change here, I will update later and standardize eslint rules across repos.

File: api/controllers/shabads.js [hunk 2]
```diff
@@ -438,64 +437,91 @@ const getShabad = (ShabadIDQ, sinceDate = null) =>
...
+                output.shabads[counter].push(row);
+              });
+
+              const outputShabadPromises = output.shabads.map(getShabadSingle);
+              output.shabads = await Promise.all(outputShabadPromises);
+              resolve(output);
             } else {
               resolve({});
             }
+            if (conn) conn.end();
```
Comment (tsingh777) [inline api/controllers/shabads.js]: nit - move this to `.finally`?

File: api/controllers/shabads.js [hunk 3]
```diff
@@ -438,64 +437,91 @@ const getShabad = (ShabadIDQ, sinceDate = null) =>
...
+                if (row.ShabadID !== curShabadID) {
+                  curShabadID = row.ShabadID;
+                  output.shabadIds.push(row.ShabadID);
+                  counter = output.shabads.push([]) - 1;
+                }
+                output.shabads[counter].push(row);
+              });
+
+              const outputShabadPromises = output.shabads.map(getShabadSingle);
+              output.shabads = await Promise.all(outputShabadPromises);
```
Comment (tsingh777) [inline api/controllers/shabads.js]: is this needed? `getShabadSingle` doesn't actually return a promise.

Comment (rajkambo) [reply api/controllers/shabads.js]: I think it's needed, because of the async call to `getShabadNavigation`

File: api/controllers/shabads.js [hunk 4]
```diff
@@ -438,64 +437,91 @@ const getShabad = (ShabadIDQ, sinceDate = null) =>
...
-              const shabadInfo = {
-                shabadId: rows[0].ShabadID,
-                pageNo: rows[0].PageNo,
-                source: lib.getSource(rows[0]),
-                raag: lib.getRaag(rows[0]),
-                writer: lib.getWriter(rows[0]),
+          .then(async rows => {
+            if (rows.length > 0 && ShabadIDQLength === 1) {
+              // single shabad
+              resolve(getShabadSingle(rows));
```
Comment (rajkambo) [inline api/controllers/shabads.js]: This looks kind of weird to me and kinda surprised it works? like resolving a promise with a promise? I guess it should work, but maybe a more readable way would be:

const retShabad = await getShabadSingle(rows);
resolve(retShabad);

File: api/controllers/shabads.js [hunk 5]
```diff
@@ -438,64 +437,91 @@ const getShabad = (ShabadIDQ, sinceDate = null) =>
...
-                  conn.end();
-                })
-                .catch(err => reject(err));
+              let curShabadID = -1;
+              let counter = 0;
+              rows.forEach(row => {
+                if (row.ShabadID !== curShabadID) {
+                  curShabadID = row.ShabadID;
+                  output.shabadIds.push(row.ShabadID);
+                  counter = output.shabads.push([]) - 1;
```
Comment (rajkambo) [inline api/controllers/shabads.js]: we want this counter initialized to -1?

**Comment by rajkambo** [comment]
Overall I'm not too sure about the context of this change, but doesn't seem like anything too crazy going on with the JS so if it works, you should be ok 👍

File: api/controllers/shabads.js (hunk 1 above)
Comment (ManjotS) [reply api/controllers/shabads.js]: yeah I was copying whats done elsewhere in the same file and repo

File: api/controllers/shabads.js (hunk 2 above)
Comment (ManjotS) [reply api/controllers/shabads.js]: theres no try to do that, so not sure how

File: api/controllers/shabads.js (hunk 5 above)
Comment (ManjotS) [reply api/controllers/shabads.js]: yes so we know its not valid and its not actually a counter, its set by push()

File: api/controllers/shabads.js (hunk 2 above)
Comment (tsingh777) [reply api/controllers/shabads.js]: promises have a `.finally`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

**Pull request merged**